programs/leap-util/actions/bls.cpp

```diff
@@ -0,0 +1,125 @@
+#include "bls.hpp"
+
+#include <fc/crypto/bls_private_key.hpp>
+#include <fc/crypto/bls_public_key.hpp>
+#include <fc/crypto/bls_signature.hpp>
+
+#include <boost/program_options.hpp>
+
+using namespace fc::crypto::blslib;
+namespace bpo = boost::program_options;
+using bpo::options_description;
+
+void bls_actions::setup(CLI::App& app) {
+   // callback helper with error code handling
+   auto err_guard = [this](int (bls_actions::*fun)()) {
+      try {
+         int rc = (this->*fun)();
+         if(rc) throw(CLI::RuntimeError(rc));
+      } catch(...) {
+         print_exception();
+         throw(CLI::RuntimeError(-1));
+      }
+   };
+
+   // main command
+   auto* sub = app.add_subcommand("bls", "BLS utility");
+   sub->require_subcommand();
+
+   // Create subcommand
+   auto create = sub->add_subcommand("create", "Create BLS items");
+   create->require_subcommand();
+
+   // sub-subcommand - key 
+   auto* create_key = create->add_subcommand("key", "Create a new BLS keypair and print the public and private keys")->callback([err_guard]() { err_guard(&bls_actions::create_key); });
+   create_key->add_option("-f,--file", opt->key_file, "Name of file to write private/public key output to. (Must be set, unless \"--to-console\" is passed");
+   create_key->add_flag( "--to-console", opt->print_console, "Print private/public keys to console.");
+
+   // sub-subcommand - pop (proof of possession) 
+   auto* create_pop = create->add_subcommand("pop", "Create proof of possession of the corresponding private key for a given public key")->callback([err_guard]() { err_guard(&bls_actions::create_pop); });
+   create_pop->add_option("-f,--file", opt->key_file, "Name of file storing the private key. (one and only one of \"-f,--file\" and \"--private-key\" must be set)");
+   create_pop->add_option("--private-key", opt->private_key_str, "The private key. (one and only one of \"-f,--file\" and \"--private-key\" must be set)");
+}
+
+int bls_actions::create_key() {
+   if (opt->key_file.empty() && !opt->print_console) {
+      std::cerr << "ERROR: Either indicate a file using \"--file\" or pass \"--to-console\"" << std::endl;
```

Comment:
`\"-f, --file\"` (add `-f` to the message)

Reply:
Thanks!